[SRVCOM-2368] Serverless multitenant istio docs (#68618)
openshift/openshift-docs: modules/serverless-ossm-installing-and-configuring-openshift-serverless.adoc

After installing {SMProductShortName}, you need to install {ServerlessProductShortName} with a specific configuration.

.Procedure

. Install Knative Serving with the following `KnativeServing` custom resource, which enables the Istio integration:
+
.Example `knative-serving-config.yaml` configuration file
[source,yaml]
----
apiVersion: operator.knative.dev/v1beta1
kind: KnativeServing
metadata:
  name: knative-serving
  namespace: knative-serving
spec:
  ingress:
    istio:
      enabled: true <1>
  deployments: <2>
  - name: activator
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: autoscaler
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  config:
    istio: <3>
      gateway.knative-serving.knative-ingress-gateway: istio-ingressgateway.<your-istio-namespace>.svc.cluster.local
      local-gateway.knative-serving.knative-local-gateway: knative-local-gateway.<your-istio-namespace>.svc.cluster.local
----
<1> Enable Istio integration.
<2> Enable sidecar injection for Knative Serving data plane pods.

> <3> If your istio is not running in the `istio-system` namespace, you need to set these two flags with the correct namespace.

Reviewer: For a follow-up:
```suggestion
<3> If your Istio is not running in the `istio-system` namespace, you must set these two flags with the correct namespace.
```
I think re: capitalization?

. Apply the `KnativeServing` resource:
+
[source,terminal]
----
$ oc apply -f knative-serving-config.yaml
----

> . Install Knative Eventing with the following `KnativeEventing` custom resource, which enables the Istio integration:

Reviewer: ```suggestion
. Install Knative Eventing with the following `KnativeEventing` object, which enables the Istio integration:
```
Again, just for consistency. "Custom resource" isn't really necessary to call out here, I don't think.

      enabled: true
  workloads: <2>
  - name: kafka-controller
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: kafka-broker-receiver
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: kafka-broker-dispatcher
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: kafka-channel-receiver
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: kafka-channel-dispatcher
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: kafka-source-dispatcher
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
  - name: kafka-sink-receiver
    annotations:
      "sidecar.istio.io/inject": "true"
      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
----
<1> The Apache Kafka cluster URL, for example `my-cluster-kafka-bootstrap.kafka:9092`.
<2> Enable sidecar injection for Knative Kafka pods.

> . Apply the `KnativeEventing` resource:

Reviewer: ```suggestion
. Apply the `KnativeEventing` object:
```
I'll stop highlighting any "resource" vs "object" discrepancies, and leave it to you to decide which to stick with throughout the PR.

+
[source,terminal]
----
$ oc apply -f knative-kafka-config.yaml
----

> . Install `ServiceEntry` to inform {SMProductShortName} of the communication between `KnativeKafka` components and an Apache Kafka cluster:

Reviewer: For a follow-up:
```suggestion
. Create a `ServiceEntry` object to inform {SMProductShortName} of the communication between `KnativeKafka` components and an Apache Kafka cluster:
```
Maybe?
